File: back/src/main/java/com/back/domain/book/dto/BookDto.java
package com.back.domain.book.dto;

import com.back.domain.book.entity.Book;
import jakarta.validation.constraints.NotNull;

public record BookDto(
        @NotNull
        Long id,
        @NotNull
        String title,
        @NotNull
        String imgUrl,
        @NotNull
        Double averageRating
) {
    public BookDto(Book book) {
        this(book.getId(), book.getTitle(), book.getImgUrl(), book.getAverageRating());
    }

}

File: back/src/main/kotlin/com/back/domain/book/dto/BookDto.kt
package com.back.domain.book.dto

import com.back.domain.book.entity.Book

open class BookDto(
    id: Long,
    title: String,
    imgUrl: String?,
    val averageRating: Double
) : BookDtoBase(id, title, imgUrl) {
    constructor(book: Book) : this(
        book.id,
        book.title,
        book.imgUrl,
        book.averageRating)
}